services/data/src/react/hooks/useDataQuery.ts

```diff
@@ -103,7 +104,19 @@ export const useDataQuery = (
         }
 
         if (newVariables) {
-            setVariables({ ...variables, ...newVariables })
+            const merged = { ...variables, ...newVariables }
+            const identical =
+                stableValueHash(variables) === stableValueHash(merged)
+
+            setVariables(merged)
```

Comment:
This should be moved into an `else` clause after `if (identical)` so that it doesn't trigger an unnecessary refresh (and subsequent serialized comparison) when we already know they're identical

Reply:
It isn't possible/allowed for the variables to contain non-serializable values right? Because if it is I can imagine we could run into trouble (the comparison would miss changes to non-serializable values).

Reply:
Fixed in e96d9848bc036e51f470721195e05f56e99adb17

Reply:
I don't think we explicitly disallow them... but I've never seen it in practice